allennlp/common/util.py

```diff
@@ -139,6 +139,32 @@ def lazy_groups_of(iterable: Iterable[A], group_size: int) -> Iterator[List[A]]:
             break
 
 
+def lazy_groups_of_max_size(
+    iterable: Iterable[A], sizes: Iterable[int], max_size: int
+) -> Iterator[List[A]]:
+    cur_size = 0
+    group: List[A] = []
+
+    iterator = iter(iterable)
+    size_iter = iter(sizes)
+
+    while True:
+        item, size = next(iterator, None), next(size_iter, None)
```

Comment:
Is this different from `for item, size in zip(iterator, size_iter):`?

Reply:
The reason I did this was that I need to finalize the last batch after the iterators are done and I wanted to just have a single loop. I rewrote it with zip and ended up finding a much neater way to create the last batch.